src/main/kotlin/com/ampnet/identityservice/service/impl/UserServiceImpl.kt

```diff
@@ -54,7 +54,7 @@ class UserServiceImpl(
 
     @Transactional
     override fun createUser(address: String): UserResponse {
-        val user = userRepository.findByAddress(address) ?: kotlin.run {
+        val user = userRepository.findByAddress(address.lowercase()) ?: kotlin.run {
             logger.info { "User is created for address: $address" }
             userRepository.save(User(address))
```

Comment:
userRepository.save(User(address.lowercase())), if use is not found you are creating it without lowercase address

Reply:
Actually not because of the User class constructor but good catch. I'll change it to be more clear for reading the code.